components/html-char-code/HtmlCharCode.tsx

```diff
@@ -0,0 +1,62 @@
+import Box from '@mui/material/Box';
+import { decode, encode } from 'html-entities';
+import React from 'react';
+
+import useLocalState from '../../hooks/useLocalState';
+import TextFieldWithCopyPaste from '../TextFieldWithCopyPaste';
+
+export default function HtmlCharCodes() {
+  const [unescaped, setUnescaped] = useLocalState<string>({
+    key: 'htmlCharCode-unescaped',
+    defaultValue: '',
+  });
+  const [escaped, setEscaped] = useLocalState<string>({
+    key: 'htmlCharCode-escaped',
+    defaultValue: '',
+  });
+
+  const clearBoth = () => {
+    setUnescaped('');
+    setEscaped('');
+  };
+
+  return (
+    <Box
+      display='flex'
+      flexDirection='column'
+      justifyContent='stretch'
+      gap={6}
+      width={1000}
+      maxWidth='100%'
+    >
+      <TextFieldWithCopyPaste
+        hasCopy
+        label='Unescaped'
+        value={unescaped}
+        onClearClick={clearBoth}
+        onPasteCleck={(text) => {
+          setUnescaped(text);
+          setEscaped(encode(text, { mode: 'extensive' }));
+        }}
+        onChange={(event) => {
+          setUnescaped(event.target.value);
+          setEscaped(encode(event.target.value, { mode: 'extensive' }));
+        }}
+      />
+      <TextFieldWithCopyPaste
+        hasCopy
+        label='Escaped'
+        value={escaped}
+        onClearClick={clearBoth}
+        onPasteCleck={(text) => {
+          setEscaped(text);
+          setUnescaped(decode(text));
+        }}
+        onChange={(event) => setEscaped(event.target.value)}
+        onBlur={(e) => {
```

Comment:
Let's move this to `onChange` so both inputs work the same. If needed we can add user facing error message when the input is causing an error.

Reply:
I was mistakenly thinking that onChange would cause the other onChange to fire.
But there was no problem, so we fixed it!